dplock/dplock.go

```diff
@@ -130,40 +130,44 @@ func (l *Lock) Acquire(ctx context.Context, id string) (lockID string, err error
 	for {
 		lockID, err = l.Lock(ctx, id)
 		if err != lock.ErrAlreadyLocked {
-			return lockID, err
+			return lockID, err // Successful or failed due to some generic error, no retry is attempted
 		}
 		if retries >= AcquireMaxRetries {
-			return "", ErrAcquireMaxRetries
+			return "", ErrAcquireMaxRetries // Failed too many times
 		}
 		retries++
 		select {
 		case <-time.After(AcquirePeriod):
-			continue
+			continue // Retry
 		case <-l.CloserChannel:
 			log.Info(ctx, "stop acquiring lock. Mongo db is being closed")
-			return "", ErrMongoDbClosing
+			return "", ErrMongoDbClosing // Abort because the app is closing
 		}
 	}
 }
 
 // Unlock releases an exclusive mongoDB lock for the provided id (if it exists)
-func (l *Lock) Unlock(ctx context.Context, lockID string) error {
+func (l *Lock) Unlock(ctx context.Context, lockID string) {
 	retries := 0
 	for {
 		_, err := l.Client.Unlock(ctx, lockID)
 		if err == nil {
-			return nil
+			if retries > 0 {
+				log.Info(ctx, "unlocking succeeded after some retries", log.Data{"retries": retries})
+			}
+			return // Successful unlock
 		}
 		if retries >= UnlockMaxRetries {
-			return ErrUnlockMaxRetries
+			log.Error(ctx, "error unlocking", ErrUnlockMaxRetries)
+			return // Failed too many times
```

Comment:
I think we should still be returning an error: ErrUnlockMaxRetries

Just in case the caller wants to take further action in the result of an unlock failure.